Jenkins/update_ci_win_docker_image.ps1

```diff
@@ -0,0 +1,83 @@
+
+$ErrorActionPreference = "Stop"
+
+$globalVariables = (Resolve-Path "$PSScriptRoot\..\global-variables.ps1").Path
+$ciUtils = (Resolve-Path "$PSScriptRoot\..\Modules\CIUtils").Path
+
+Import-Module $ciUtils
+. $globalVariables
+
+$WORKSPACE="$env:WORKSPACE"
+$DOCKER_HUB_USER="$env:DOCKER_HUB_USER"
+$DOCKER_HUB_USER_PASSWORD="$env:DOCKER_HUB_USER_PASSWORD"
+$DOCKERFILE_PATH="$PSScriptRoot\ci-image-Dockerfile"
+$IMAGE_FILES_DIR="$WORKSPACE\files"
+$IMAGE_FILES_BASE_URL="$LOG_SERVER_BASE_URL/downloads"
+
+if (!$DOCKER_HUB_USER) {
+    Write-Output "Docker user is not set."
+    exit 1
+}
+
+if (!$DOCKER_HUB_USER_PASSWORD) {
+    Write-Output "Docker user password is not set."
+    exit 1
+}
+
+# Do a login into docker
+Write-Output $DOCKER_HUB_USER_PASSWORD | docker login -u $DOCKER_HUB_USER --password-stdin
+
+pushd $WORKSPACE
+
+# Cleanup the workspace and docker images
+if (Test-Path -Path $WORKSPACE\*) {
+    Write-Output "Workspace directory not empty."
+    Remove-Item -Recurse -Force "$WORKSPACE\*"
+}
+
+if (docker image list ${DOCKER_HUB_USER}/windows -q -a) {
```

Comment:
Get the image list via `Start-ExternalCommand` because we want to check the `$LASTEXITCODE` for it.

This line should translate to:
```
$imageList = Start-ExternalCommand -ScriptBlock { docker image list ${DOCKER_HUB_USER}/windows -q -a } `
                                   -ErrorMessage "Failed to run docker image list"
if (!$imageList) {
...
...
```

Please do the same on all if's below.